pkg/aws/eni/node_manager_test.go

```diff
@@ -520,15 +530,15 @@ func (e *ENISuite) TestNodeManagerExceedENICapacity(c *check.C) {
 	c.Assert(node.Stats().AvailableIPs, check.Equals, 20)
 	c.Assert(node.Stats().UsedIPs, check.Equals, 0)
 
-	// Use 41 out of 45 IPs, we should reach 4 addresses needed but never 0 addresses needed
-	mngr.Update(updateCiliumNode(cn, 45, 41))
-	c.Assert(testutils.WaitUntil(func() bool { return reachedAddressesNeeded(mngr, "node2", 4) }, 5*time.Second), check.IsNil)
-	c.Assert(testutils.WaitUntil(func() bool { return reachedAddressesNeeded(mngr, "node2", 0) }, 5*time.Second), check.Not(check.IsNil))
+	// Use 40 out of 40 available IPs, we should reach 0 address needed once
+	// we assigned the remaining five that the m4.xlarge instance type supports
+	mngr.Update(updateCiliumNode(cn, 40, 40))
+	c.Assert(testutils.WaitUntil(func() bool { return reachedAddressesNeeded(mngr, "node2", 0) }, 5*time.Second), check.IsNil)
```

Comment:
/cc @tgraf 

I'm not familiar with these tests but I found it a bit confusing that this comment says "out of 40 available IPs" but the assert below still asserts that `node.Stats().AvailableIPs` equals 45.

Reply:
I attempted to make it easier to understand by updating the test values and the comment. Basically the scenario is now:
- the instance type is an `m4.xlarge` which can theorically support **45** addresses
- 42 addresses are already allocated to the node (`available`)
- 40 addresses are `used`

`PreAllocate` being set to **8**, we expect IPAM to:
- Allocate the remaining 3 addresses of the node (max [45] - available [42])
- Remain with 40 used IPs
- Return **0** addresses needed given that we have reached the maximum set on the node